[Keyboard] Add mini42

controller-works opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->
Add support for mini42 split ergonomic keyboard with RP2040. Includes default and via keymaps.

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented on keyboards/controllerworks/mini42/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,175 @@
+};
+
+#ifdef OLED_ENABLE
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
```
Ideally, this should be moved to the keyboard's c file, using `oled_(init|task)_kb`, so it provides a default behavior without needing it to be specified at the keymap level, and can be overridden.

controller-works replied:
I have moved it to the c file. I am not sure about making it so it can be overridden.

drashna commented on keyboards/controllerworks/mini42/readme.md:
```diff
@@ -0,0 +1,77 @@
+# mini42
+
+![mini42](https://i.imgur.com/xrSKhqR.png)
+![mini42PCB](https://i.imgur.com/R7u1KB8.png)
```
```suggestion
![mini42](https://i.imgur.com/xrSKhqRh.png)
![mini42PCB](https://i.imgur.com/R7u1KB8h.png)
```

drashna commented on keyboards/controllerworks/mini42/rules.mk:
```diff
@@ -0,0 +1,13 @@
+LTO_ENABLE = yes
```
Most of these can be moved to the info.json (everything but the "driver" option)

controller-works replied:
I have now moved everything I could find into the json file.

controller-works commented on keyboards/controllerworks/mini42/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,175 @@
+  }
+
+  // update keylog
+  snprintf(keylog_str, sizeof(keylog_str), "%dx%d, k%2d : %c",
```
This has been removed.

controller-works commented on keyboards/controllerworks/mini42/config.h:
```diff
@@ -0,0 +1,75 @@
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+#define OLED_FONT_H "keyboards/controllerworks/mini42/lib/glcdfont.c"
```
Done.

controller-works commented:
I have almost completely rewritten the keyboard files to use the data driven json file. I also simplified the OLED behavior and moved it to the keyboard c file. I also removed the customized font. I am not sure if I need to do something for you to see the changes in the mini42 branch of my fork, or if this happens automatically. This is my first PR. Thanks for taking the time to review this.

drashna commented on keyboards/controllerworks/mini42/config.h:
```diff
@@ -0,0 +1,70 @@
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+#define OLED_FONT_H "drivers/oled/glcdfont.c"
```
This can be removed, as it is default. 
```suggestion
```

drashna commented on keyboards/controllerworks/mini42/mini42.c:
```diff
@@ -0,0 +1,81 @@
+    return false;
+}
+
+bool oled_task_user(void) {
```
```suggestion
bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return false;
    }
```

controller-works replied:
Done.

controller-works commented:
@Drashna it looks like all of the checks are passing.

controller-works reviewed (commented):
I think all suggested changes have been made.

drashna commented on keyboards/controllerworks/mini42/mini42.c:
```diff
@@ -0,0 +1,84 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "mini42.h"
+#include "split_util.h"
```
```suggestion
#include "mini42.h"
```

drashna commented on keyboards/controllerworks/mini42/mini42.c:
```diff
@@ -0,0 +1,84 @@
+    oled_write_raw_P(raw_logo, sizeof(raw_logo));
+}
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
```
Sorry, I missed this earlier
```suggestion
oled_rotation_t oled_init_kb(oled_rotation_t rotation) {
```
